lab/tree_builder.py

```diff
@@ -15,19 +17,49 @@ class TreeBuilder(object):
     def __init__(self, root):
         # root will be type TreeNode
         self.root = root
-        self.size = 0
+        self.size = 1
 
-    # Function to push to tree 
+    # Function to push to tree
     def push(self, node):
-        # TODO
-        pass
+        # pointer for the current node
+        currNode = self.root
+        while True:
+            # if value of node to be added is greater, go right
+            if node.val > currNode.val:
+                # if right child exists, go right
+                if currNode.right:
+                    currNode = currNode.right
+                # if no right child, add node as right child
+                else:
+                    currNode.right = node
+                    node.parent = currNode
+                    break
+            # else, go left
+            else:
+                # if left child exists, go left
+                if currNode.left:
+                    currNode = currNode.left
+                # if no left child, add node as left child and break
+                else:
+                    currNode.left = node
+                    node.parent = currNode
+                    break
 
-    # Function to remove node from tree 
+        # increment size of tree on pushing new node
+        self.size += 1
+        return
+
+    # Function to remove node from tree
     def pop(self, name):
         # TODO
         pass
 
     # Function to export tree as png(Use DotExporter)
     def export(self, file="tree"):
-        # TODO 
-        pass
\ No newline at end of file
+
+        # export to dot file
+        DotExporter(self.root).to_dotfile(file+".dot")
+        Source.from_file(file+'.dot')
+        # render as png
+        render('dot', 'png', file+'.dot')
```

Comment:
This line is throwing an error for me when I trying to convert dot file to png. 
```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "visualize.py", line 31, in <module>
    main()
  File "visualize.py", line 27, in main
    tree.export()
  File "/Users/sharvilkekre/projects/NBA-Search/lab/tree_builder.py", line 64, in export
    render('dot', 'png', file+'.dot')
  File "/anaconda3/envs/home/lib/python3.7/site-packages/graphviz/backend.py", line 221, in render
    run(cmd, capture_output=True, cwd=cwd, check=True, quiet=quiet)
  File "/anaconda3/envs/home/lib/python3.7/site-packages/graphviz/backend.py", line 167, in run
    raise ExecutableNotFound(cmd)
graphviz.backend.ExecutableNotFound: failed to execute ['dot', '-Tpng', '-O', 'tree.dot'], make sure the Graphviz executables are on your systems' PATH
```

I can however see that the dot file is being successfully created and written to disk. The only issue lies in conversion from dot to png.

Reply:
There may be a different way to convert dot to png with `graphviz` or perhaps maybe even think about using the `subprocess` module
```python
from subprocess import check_call
check_call(['dot','-Tpng','InputFile.dot','-o','OutputFile.png'])
```

Reply:
Hi @skekre98
Yup I am familiar with this error since I was getting that as well. The `DotExporter` has a dependency on `graphviz` anyway so it will be required.
I got my segment of the code working by using both `pip install graphviz` and `conda install graphviz`, followed by putting `Anaconda/bin/graphviz` in PATH (since the brave issue is thrown due to issues with the PATH)

Reply:
Is this something you would be able to do in the setup script? Also what is user in not using anaconda? Perhaps it may be easier to install a python package that can handle it as standalone.